gadopt/inverse.py

```diff
@@ -359,3 +360,54 @@ def __getstate__(self):
         "Iteration Limit": 100,
     },
 }
+
+
+class DiagnosticBlock(Block):
```

Comment:
Docstring: What/why is this block?

Reply:
I've added some docstrings. It was from thwaites basicailly a helper block to allow you to output gradients wrt to function provided. It was useful for time dependent simulations (and also for visualising gradient at different iteration of optimisation alogrithm)